### PR TITLE
Feature/apply sequenceT

### PR DESCRIPTION
**Pull request by raveclassic**
#421

### CODE REVIEW

File: src/Apply.ts [hunk 1]
```diff
@@ -246,3 +246,107 @@ export function getSemigroup<F, A>(F: Apply<F>, S: Semigroup<A>): () => Semigrou
...
+ * @param F - {@link Apply} instance
+ */
+export function sequenceT<F extends URIS3, U, L>(F: Apply3<F>): SequenceT3<F>
+export function sequenceT<F extends URIS3, U, L>(F: Apply3C<F, U, L>): SequenceT3C<F, U, L>
+export function sequenceT<F extends URIS2>(F: Apply2<F>): SequenceT2<F>
+export function sequenceT<F extends URIS2, L>(F: Apply2C<F, L>): SequenceT2C<F, L>
+export function sequenceT<F extends URIS>(F: Apply1<F>): SequenceT1<F>
+export function sequenceT<F>(F: Apply<F>): SequenceT<F> {
+  return (...list: any[]) => {
+    const args = list.reverse()
```
Comment (gcanti) [inline src/Apply.ts]: Ops sorry, I didn't notice that my results were reversed. We can avoid `list.reverse()` by changing the order in the `for` loop

```ts
export function sequenceT<F>(F: Apply<F>): SequenceT<F> {
  return (...args: any[]) => {
    const len = args.length
    let f = tupleConstructors[len]
    if (!Boolean(f)) {
      f = tupleConstructors[len] = curried((...args: any[]): any[] => args, len - 1, [])
    }
    let r = F.map(args[0], f)
    for (let i = 1; i < len; i++) {
      r = F.ap(r, args[i])
    }
    return r
  }
}
```

File: src/Apply.ts [hunk 2]
```diff
@@ -246,3 +246,107 @@ export function getSemigroup<F, A>(F: Apply<F>, S: Semigroup<A>): () => Semigrou
...
+ * @function
+ * @since 1.5.0
+ * @param F - {@link Apply} instance
+ */
+export function sequenceT<F extends URIS3, U, L>(F: Apply3<F>): SequenceT3<F>
+export function sequenceT<F extends URIS3, U, L>(F: Apply3C<F, U, L>): SequenceT3C<F, U, L>
+export function sequenceT<F extends URIS2>(F: Apply2<F>): SequenceT2<F>
+export function sequenceT<F extends URIS2, L>(F: Apply2C<F, L>): SequenceT2C<F, L>
+export function sequenceT<F extends URIS>(F: Apply1<F>): SequenceT1<F>
+export function sequenceT<F>(F: Apply<F>): SequenceT<F> {
```
Comment (gcanti) [inline src/Apply.ts]: We must also export the base signature here

```ts
export function sequenceT<F>(F: Apply<F>): SequenceT<F>
```

File: src/Apply.ts (hunk 1 above)
Comment (raveclassic) [reply src/Apply.ts]: nice, done

File: src/Apply.ts (hunk 2 above)
Comment (raveclassic) [reply src/Apply.ts]: You mean to duplicate it? ok, done

**Comment by gcanti** [comment]
@raveclassic nice, thanks. Could you please squash the commits while merging?

**Pull request merged**